src/ASM/LR/ASMu2Dmx.C

```diff
@@ -201,9 +201,15 @@ bool ASMu2Dmx::generateFEMTopology ()
   if (!myMLGN.empty())
     return true;
 
+  auto createLR = [this](Go::SplineSurface& srf)
+                  {
+                    return srf.rational() ? this->createLRNurbs(srf)
+                                          : new LR::LRSplineSurface(&srf);
+                  };
+
   if (tensorPrjBas)
   {
-    projBasis.reset(new LR::LRSplineSurface(tensorPrjBas));
+    projBasis.reset(createLR(*tensorPrjBas));
```

Comment:
Here you first dereference the tensorPrjBas pointer, before you (inside the lambda) use & to extract the pointer again (back and forth), wouldn't in this case at least be better to let the lambda accept the pointer argument instead? Since it is local for this method. In spite of your policy of using reference whenever the pointer can't be NULL.

Reply:
I did initially, but it comes at the cost of .get() for the shared_ptrs. So i went with the least visual noise, deref and one ref (for the evil lr constructor that takes a ptr where null makes no sense 😁)

Reply:
And lr even takes a mutable pointer hence i cant pass a const ref :/